src/mappening/api/utils/location_utils.py

```diff
@@ -0,0 +1,256 @@
+from mappening.utils.database import *
+from mappening.api.utils import tokenize
+
+from flask import Flask, jsonify, request, json, Blueprint
+from flask_cors import CORS, cross_origin
+import requests
+import re
+import json
+import os
+from operator import itemgetter
+
+# Latitude and Longitude range from (-90, 90) and (-180, 180)
+INVALID_COORDINATE = 420
+
+# Set parameters for Bruin Bear/Center of UCLA
+CENTER_LATITUDE = "34.070966"
+CENTER_LONGITUDE = "-118.445"
+
+# For comprehension: School of Theater, Film, TV within radius 700
+# Hammer Museum within radius 1300, Saffron and Rose within radius 1800
+RADIUS = "2000"
+
+# Go through all events in given events db and extract unique locations from the events
+# Return the array of locations discovered
+def get_locations_from_collection(events_collection):
+    # Iterate through all events and get list of unique venues
+    places = []
+    place = {}
+
+    if events_collection == "ucla_events":
+      events_cursor = ucla_events_collection.find({"place": {"$exists": True}})
+    elif events_collection == "events_ml":
+      events_cursor = events_ml_collection.find({"place": {"$exists": True}})
+    else: # events_collection == "test":
+      events_cursor = test_collection.find({"place": {"$exists": True}})
+    
+    # Every time there are new events, check location info and update db if necessary
+    # events_cursor = events_ml_collection.find({"place": {"$exists": True}})
+    # events_cursor = events_collection.find({"place": {"$exists": True}})
+    if events_cursor.count() > 0:
+      for event in events_cursor:
```

Comment:
This chain of if statements is p disgusting, consider separating into multiple functions/sub functions and calling those

Reply:
Heh no it's super gross but right now just keeping it in a "working-but-really-need-to-fix" stage

Reply:
Todo